libs/api-client-framework/src/endpoint.rs

```diff
@@ -0,0 +1,55 @@
+use serde::{Deserialize, Serialize};
+use std::{borrow::Cow, fmt::Debug};
+use url::Url;
+
+/// Represents a specification for an API call that can be built into an HTTP request and sent.
+/// New endpoints should implement this trait.
+///
+/// If the request succeeds, the call will resolve to a `Response`.
+pub trait Endpoint {
```

Comment:
any way to define custom headers? to pass api key to endpoint for example

Reply:
btw found this implementation https://github.com/hephex/api/blob/master/src/lib.rs#L162 , Im not sure we want to use it but take a look at different implementation is good practice

Reply:
I suggest to add custom headers later when we are required. Agree that it could be useful, but prefer to consider it as mvp for now